Stratis.Bitcoin.Tests/BlockStore/LoopTests/BlockStoreLoopStepCheckNextChainedBlockExistStepTest.cs

```diff
@@ -13,29 +13,30 @@ public sealed class BlockStoreLoopStepCheckNextChainedBlockExistStepTest : Block
         [Fact]
         public void CanExecute_CheckNextChainedBlockExistStep()
         {
-            List<Block> blocks = CreateBlocks(5);
+            var blocks = CreateBlocks(5);
 
-            // The BlockRepository has 5 blocks stored
-            using (var blockRepository = new BlockRepository(Network.Main, TestBase.AssureEmptyDirAsDataFolder(@"BlockStore\LoopTest_Exists")))
+            using (var blockRepository = new BlockRepository(Network.Main, TestBase.AssureEmptyDirAsDataFolder(@"BlockStore\Exists")))
             {
+                // Push 5 blocks to the repository
                 blockRepository.PutAsync(blocks.Last().GetHash(), blocks).GetAwaiter().GetResult();
 
-                var chain = new ConcurrentChain(Network.Main);
-
                 // The chain has 4 blocks appended
-                AppendBlocks(chain, blocks.Take(4));
+                var chain = new ConcurrentChain(blocks[0].Header);
+                AppendBlocks(chain, blocks.Skip(1).Take(3));
 
                 // Create the last chained block without appending to the chain
-                ChainedBlock block03 = chain.GetBlock(blocks[3].GetHash());
-                var block04 = new ChainedBlock(blocks[4].Header, blocks[4].Header.GetHash(), block03);
+                var block03 = chain.GetBlock(blocks[3].GetHash());
+                if (block03.Header.HashPrevBlock != blocks[4].Header.HashPrevBlock)
+                    Console.WriteLine(string.Format("{0} - {1} - {2}", block03.Header.HashPrevBlock, blocks[4].Header.HashPrevBlock, chain.Genesis.HashBlock));
 
-                BlockStoreLoop blockStoreLoop = CreateBlockStoreLoop(chain, blockRepository, @"BlockStore\LoopTest_Exists");
+                var block04 = new ChainedBlock(blocks[4].Header, blocks[4].Header.GetHash(), block03);
 
+                var blockStoreLoop = CreateBlockStoreLoop(chain, blockRepository, @"BlockStore\Exists");
```

Comment:
I don't this is correct here because if you call AssureEmptyDirAsDataFolder it'll clear the repository from the blocks you inserted a few lines up:

```
// Push 5 blocks to the repository
blockRepository.PutAsync(blocks.Last().GetHash(), blocks).GetAwaiter().GetResult();
```